### PR TITLE
openshift: 1.5.0 -> 3.6.0

### PR DESCRIPTION
**Pull request by moretea**
<s>Unfortunately, `oc cluster up` still does not work after this update (but neither did it work on 1.5.0)</s>

- Updated from 1.5.0 to 3.6.0 (this is just the next version, but Red Hat did quite the version bump there)
- Added 'v' to the version; it is used by `oc cluster up` to determine  which image should be downloaded.
- Added myself as a maintainer.

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by moretea** [comment]
@offlinehacker @bachp could you take a look?

**Comment by bachp** [comment]
Changes look good to me. I never tried `oc cluster up` on NixOS so far but I like that you were able to get
 it working 👍

File: pkgs/applications/networking/cluster/openshift/default.nix [hunk 1]
```diff
@@ -14,21 +14,32 @@ in buildGoPackage rec {
...
+      --replace 'nsenter --mount=/rootfs/proc/1/ns/mnt findmnt' \
+      'nsenter --mount=/rootfs/proc/1/ns/mnt ${utillinux}/bin/findmnt'
+
+    substituteInPlace pkg/bootstrap/docker/host/host.go  \
+      --replace 'nsenter --mount=/rootfs/proc/1/ns/mnt mount' \
+      'nsenter --mount=/rootfs/proc/1/ns/mnt ${utillinux}/bin/mount'
+
+    substituteInPlace pkg/bootstrap/docker/host/host.go  \
+      --replace 'nsenter --mount=/rootfs/proc/1/ns/mnt mkdir' \
+      'nsenter --mount=/rootfs/proc/1/ns/mnt ${coreutils}/bin/mount'
```
Comment (Mic92) [inline pkgs/applications/networking/cluster/openshift/default.nix]: mount is in util-linux only

File: pkgs/applications/networking/cluster/openshift/default.nix [hunk 2]
```diff
@@ -14,21 +14,32 @@ in buildGoPackage rec {
...
   };
 
   buildInputs = [ which ];
 
   goPackagePath = null;
   patchPhase = ''
     patchShebangs ./hack
+    substituteInPlace pkg/bootstrap/docker/host/host.go  \
+      --replace 'nsenter --mount=/rootfs/proc/1/ns/mnt findmnt' \
+      'nsenter --mount=/rootfs/proc/1/ns/mnt ${utillinux}/bin/findmnt'
```
Comment (Mic92) [inline pkgs/applications/networking/cluster/openshift/default.nix]: Please also make replace `nsenter` with `${utillinux}/bin/nsenter`

File: pkgs/applications/networking/cluster/openshift/default.nix (hunk 1 above)
Comment (moretea) [reply pkgs/applications/networking/cluster/openshift/default.nix]: Yes, thanks.

File: pkgs/applications/networking/cluster/openshift/default.nix (hunk 2 above)
Comment (moretea) [reply pkgs/applications/networking/cluster/openshift/default.nix]: That will break stuff! These `nsenter`'s are run from within the openshift container. It enters the namespace of the host of the container, and then executes e.g. findmt, mount & mkdir in there.

**Comment by moretea** [comment]
I think that solves all of the issues.

**Comment by bachp** [comment]
I only tested the `oc` client command, but they work fine. :+1:

**Comment by Mic92** [comment]
It was broken before right? So maybe I should backport it to stable?

**Comment by bachp** [comment]
@Mic92 The command line part, which is the only part I used so far was always working. At least I never had issues.

On the other hand having 3.6.0 in stable would also be good :)

**Pull request merged**